app/src/main/res/drawable/bg_state_button.xml

```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:drawable="@drawable/bg_primary_button" android:state_enabled="true"/>
+    <item android:drawable="@drawable/bg_transparent_button" android:state_enabled="false" />
+    <!-- default state-->
+    <item android:drawable="@drawable/bg_transparent_button" />
```

Comment:
You could simplify this by setting `bg_primary_button` as default
```
<selector xmlns:android="http://schemas.android.com/apk/res/android">
    <item android:drawable="@drawable/bg_transparent_button" android:state_enabled="false" />
    <item android:drawable="@drawable/bg_primary_button"/>
</selector>
```

Reply:
Also, I'd rename these files to be more clear what they are for:
`bg_primary_button` -> `bg_primary_button_enabled`
`bg_transparent_button` -> `bg_primary_button_disabled`
`bg_state_button` -> `bg_primary_button` or `bg_primary_button_states`